### PR TITLE
Semi finish

### PR DESCRIPTION
**Pull request by DilsherB**
screen shots:
alter table
average
avg with group by and where
Count with condtion
count
data inserted
Group By
max
savepoint with rollback 2
savepoint with rollback
transaction with commit
transaction with rollback
transaction

in this PR

1. I altered schema to have another column in my table "species".
2. I added some more data to the table "animals".

I added following queries:
1. update data by transaction and roll it back.
2. update data by transaction and commit it.
3. delete all data and rollback it.
4. delete all animals born after Jan 1st, 2022.
5. Create a savepoint for the transaction.
6. Update all animals' weight to be their weight multiplied by -1 and rollback to savepoint.
7. commit the transaction

Following aggregate queries also added.
1. `count():` find number of animals.
2. `count() with where:` number of animals never tried to escape.
3. `avg():` average weight of animals.
4. `max() with where:` Who escapes the most, neutered or not neutered animals?
5. `min() & max() with group by:` What is the minimum and maximum weight of each type of animal?
6. `avg() with where & group by:` What is the average number of escape attempts per animal type of those born between 1990 and 2000?

### CODE REVIEW

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @DilsherB, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has a link in the readme file. Kindly remove

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by DilsherB** [comment]
@Olamarx Can you please specify about which link you are talking.

**Comment by Olamarx** [comment]
> @Olamarx Can you please specify about which link you are talking.

I am talking of this

image

**Comment by DilsherB** [comment]
@Olamarx thanks

File: data.sql [hunk 1]
```diff
@@ -1,35 +1,31 @@
...
-);
+  )
+VALUES
+  ('Charmander', '2020-02-08', 0, false, 11),
+  ('Plantmon', '2021-11-15', 2, TRUE, 5.7),
+  ('Squirtle', '1993-04-02', 3, false, 12.13),
+  ('Angemon', '2005-06-12', 1, TRUE, 45),
+  ('Boarmon', '2005-06-07', 7, TRUE, 20.4),
+  ('Blossom', '1998-10-13', 3, TRUE, 17),
+  ('Ditto', '2022-05-14', 4, TRUE, 22);
```
Comment (ShadyShawkat) [inline data.sql]: - [x] Please make sure to insert the given data correctly. Some of the given values for `weight_kg` are negative, please change them to reflect that.

File: queries.sql [hunk 2]
```diff
@@ -31,4 +30,71 @@ WHERE name <> 'Gabumon';
...
+WHERE date_of_birth > '2022-01-01';
+
+SAVEPOINT DOB;
+
+UPDATE animals
+SET weight_kg = weight_kg * '-1';
+
+ROLLBACK TO SAVEPOINT DOB;
+
+COMMIT;
```
Comment (ShadyShawkat) [inline queries.sql]: - [x] Kindly make sure to write the query for [this following quest](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Update%20all%20animals%27%20weights%20that%20are%20negative%20to%20be%20their%20weight%20multiplied%20by%20%2D1.) before commiting.

File: queries.sql [hunk 3]
```diff
@@ -31,4 +30,71 @@ WHERE name <> 'Gabumon';
...
+
+SELECT COUNT(name)
+FROM animals
+WHERE escape_attempts = '0';
+
+SELECT AVG(weight_kg)
+FROM animals;
+
+SELECT MAX(escape_attempts)
+FROM animals;
```
Comment (ShadyShawkat) [inline queries.sql]: - [x] Please change this query to reflect [the given quest](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Who%20escapes%20the%20most%2C%20neutered%20or%20not%20neutered%20animals%3F). The selected data should showcase which group of animals escapes the most, neutered or not neutered animals.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @DilsherB, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (DilsherB) [reply data.sql]: Hi, @ShadyShawkat all values are positive. the attached screen shots also depicting that the values are positive not negative.

File: queries.sql (hunk 2 above)
Comment (DilsherB) [reply queries.sql]: Hi, @ShadyShawkat I think you have not paid attention to the code following is the code snippet I applied which is exactly the same as required.
image

File: queries.sql (hunk 3 above)
Comment (DilsherB) [reply queries.sql]: implemented.

File: data.sql (hunk 1 above)
Comment (ShadyShawkat) [inline data.sql]: - [ ] Please make sure to insert the given data correctly. From the following screenshot, you can see some of the given data which are negative.

image

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @DilsherB, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

Image

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (DilsherB) [reply data.sql]: @ShadyShawkat please watch this video and tell me about the issue you raised.
https://youtu.be/m2fIWhlP2OQ

**Comment by ahmetbozaci** [comment]
### Hello @DilsherB 
### _This is an additional review as a result of the second opinion request._ 

- [x] Kindly insert the correct data that is given in the [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Insert%20the%20following,escape%204%20times.).
- As you can see in the [requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md#:~:text=Insert%20the%20following,escape%204%20times.) some weights are negative values but in your code all of them are positive
image
__
**YOUR CODE**
image

___



### Cheers and Happy coding! 💻 👏
____
[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Comment by DilsherB** [comment]
@ahmetbozaci , thank you for explaining in the right way. I have implemented the required changes.

**Review by omarsalem7 — APPROVED** [review]
Hi @DilsherB 😊, 
# STATUS: APPROVED! 🥇 💯 



Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**